AlphaWallet/Activities/ViewControllers/ActivitiesViewController.swift

```diff
@@ -254,3 +284,66 @@ extension ActivitiesViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView, willDisplayHeaderView view: UIView, forSection section: Int) {
     }
 }
+
+extension ActivitiesViewController: UISearchResultsUpdating {
+    //At least on iOS 13 beta on a device. updateSearchResults(for:) is called when we set `searchController.isActive = false` to dismiss search (because user tapped on a filter), but the value of `searchController.isActive` remains `false` during the call, hence the async.
+    //This behavior is not observed in iOS 12, simulator
+    public func updateSearchResults(for searchController: UISearchController) {
+        DispatchQueue.main.async {
+            self.processSearchWithKeywords()
+        }
+    }
+
+    private func processSearchWithKeywords() {
+        viewModel.filter(.keyword(searchController.searchBar.text)) { filteredItems in
+            self.viewModel.filteredItems = filteredItems
```

Comment:
Do you think it's better we move this line into the `filter()` function, then `filterItems` can be made private?

Reply:
i made filteredItems as internal because we are using struct for `ActivitiesViewModel` and we can't capture `self` in 
`DispatchQueue.global().async {`